src/coreclr/pal/src/include/pal/mutex.hpp

```diff
@@ -124,9 +124,11 @@ Miscellaneous
 //   involved. See https://github.com/dotnet/runtime/issues/10519.
 // - On OSX, pthread robust mutexes were/are not available at the time of this writing. In case they are made available in the
 //   future, their use is disabled for compatibility.
+// - On RISCV64 (with QEMU), pthread robust mutexes were/are not available at the time of this writing. In case they are made available in the
+//   future, their use is disabled for compatibility.
 #if HAVE_FULLY_FEATURED_PTHREAD_MUTEXES && \
     HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES && \
-    !(defined(__FreeBSD__) || defined(TARGET_OSX))
+    !(defined(__FreeBSD__) || defined(TARGET_OSX) || defined(TARGET_RISCV64))
```

Comment:
I wonder if we should define the `HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES_EXITCODE` to 1 (=> we don't have them) for RISCV64 in the tryrun.cmake. Now we set it to 0 for may architectures including RISCV64  here:
https://github.com/dotnet/runtime/blob/53bb518ea482a5960577b151c11cc1297e5dea6d/eng/native/tryrun.cmake#L163-L165

This is used for cross compilation. For compiling on real RISCV6 device, it would get detected at build time using a test execution in the configure.cmake here:
https://github.com/dotnet/runtime/blob/0f1866ead5aef15d47983692bafb4ba20cd3aa98/src/coreclr/pal/src/configure.cmake#L1121-L1308

Reply:
The codes set ```HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES_EXITCODE``` to 0 to fix paltest fail (```threading/NamedMutex/test1/paltest_namedmutex_test1```).
Do you mean we should set it to 1?

Reply:
I am confused - setting the `HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES_EXITCODE` in the tryrun.cmake to 0 means that the riscv Linux has functional robust mutexes. But in your comment, you say it doesn't. That's why I have mentioned setting it to 1 for riscv4 which should do the same thing as the extra condition you have added to the `#if`. In the cross compilation case.

Reply:
I see. I will set `HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES_EXITCODE` to `1` like others.
Thank you.

Reply:
With the change in the tryrun.cmake, the change in this ifdef should not be needed (because the `HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES` is now 0)

Reply:
Thank you. Removed the codes.